File: 0234-palindrome-linked-list/0234-palindrome-linked-list.go
package main
func isPalindrome(head *ListNode) bool {
    var values []int

    // Store linked list values in a slice
    for head != nil {
        values = append(values, head.Val)
        head = head.Next
    }

    // Check if the slice is a palindrome by comparing from both ends
    for i := 0; i < len(values)/2; i++ {
        if values[i] != values[len(values)-i-1] {
            return false
        }
    }

    return true
}

File: 0234-palindrome-linked-list/0234-palindrome-linked-list.py
# Definition for singly-linked list.
# class ListNode:
#     def __init__(self, val=0, next=None):
#         self.val = val
#         self.next = next

class Solution:
    def isPalindrome(self, head: Optional[ListNode]) -> bool:
        values = []
        
        # Store linked list values in a list
        while head:
            values.append(head.val)
            head = head.next
        
        # Check if the list is a palindrome by comparing from both ends
        for i in range(len(values) // 2):
            if values[i] != values[-(i + 1)]:
                return False
        
        return True
